src/main/java/nl/tudelft/dnainator/graph/GraphQueryDescription.java

```diff
@@ -0,0 +1,169 @@
+package nl.tudelft.dnainator.graph;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.function.Predicate;
+
+import nl.tudelft.dnainator.core.SequenceNode;
+
+/**
+ * A description of a query and its parameters.
+ */
+public class GraphQueryDescription {
+	private Collection<String> idStrings;
+	private Collection<String> sourceStrings;
+	private Predicate<SequenceNode> filter;
+	private boolean queryFrom = false;
+	private int from = 0;
+	private boolean queryTo = false;
+	private int to = Integer.MAX_VALUE;
+
+	/**
+	 * Query for the given id.
+	 * @param id the String id to search for
+	 * @return this
+	 */
+	public GraphQueryDescription hasId(String id) {
+		return haveIds(Collections.singletonList(id));
+	}
+
+	/**
+	 * Query for the given ids.
+	 * @param ids the ids to search for.
+	 * @return this
+	 */
+	public GraphQueryDescription haveIds(Collection<String> ids) {
+		if (idStrings == null) {
+			idStrings = new ArrayList<>(ids);
+			return this;
+		}
+		idStrings.addAll(ids);
+		return this;
+	}
+
+	/**
+	 * @return Whether the query should query for ids.
+	 */
+	public boolean shouldQueryIds() {
```

Comment:
Yes, for nothing else. I wanted to think of a way to construct the query while the querydescription is build, but it didn't seem possible because each Graph implementation handles descriptions differently. So we need to "compile" it in the specific implementations.

These methods are handy, as a condition should only be included if it was specified 1 or more times..